File: projects/apis/metal/loader/metal_command_loader.c
#include "metal_loader.h"

gnCommandFunctions loadMetalCommandFunctions() {
    return (gnCommandFunctions){
        ._gnCommandPoolAllocateCommandBuffers = NULL,
        ._gnBeginCommandBuffer = NULL,
        ._gnResetCommandBuffer = NULL,
        ._gnEndCommandBuffer = NULL,

        ._gnCommandBeginRenderPass = NULL,
        ._gnCommandEndRenderPass = NULL,
        ._gnCommandBindGraphicsPipeline = NULL,
        ._gnCommandSetViewport = NULL,
        ._gnCommandSetScissor = NULL,
        ._gnCommandBindUniform = NULL,
        ._gnCommandPushConstant = NULL,

        ._gnCommandBindBuffer = NULL,
        ._gnCommandDraw = NULL,
        ._gnCommandDrawIndexed = NULL,
    };
}

File: projects/apis/metal/loader/metal_command_loader.m
#include "metal_loader.h"
#include "commands/command_buffer/metal_command_buffer.h"
#include "commands/commands/metal_commands.h"

gnCommandFunctions loadMetalCommandFunctions(void) {
    return (gnCommandFunctions){
        ._gnCommandPoolAllocateCommandBuffers = allocateMetalCommandBuffers,
        ._gnBeginCommandBuffer = beginMetalCommandBuffer,
        ._gnResetCommandBuffer = resetMetalCommandBuffer,
        ._gnEndCommandBuffer = endMetalCommandBuffer,
        ._gnDestroyCommandBuffer = destroyMetalCommandBuffer,

        ._gnCommandBeginRenderPass = metelBeginRenderPass,
        ._gnCommandEndRenderPass = endMetalRenderPass,
        ._gnCommandBindGraphicsPipeline = bindMetalGraphicsPipeline,
        ._gnCommandSetViewport = setMetalViewport,
        ._gnCommandSetScissor = setMetalScissor,
        ._gnCommandBindUniform = metalBindUniform,
        ._gnCommandPushConstant = metalBindVertexBytes,

        ._gnCommandBindBuffer = bindMetalBuffer,
        ._gnCommandDraw = metalDraw,
        ._gnCommandDrawIndexed = metalDrawIndexed,
    };
}
